pkg/tarsum/builder_context_test.go

```diff
@@ -0,0 +1,63 @@
+package tarsum
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+// Try to remove tarsum (in the BuilderContext) that do not exists, won't change a thing
+func TestTarSumRemoveNonExistent(t *testing.T) {
+	filename := "testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/layer.tar"
+	reader, err := os.Open(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts, err := NewTarSum(reader, false, Version0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Read and discard bytes so that it populates sums
+	_, err = io.Copy(ioutil.Discard, ts)
+	if err != nil {
+		t.Errorf("failed to read from %s: %s", filename, err)
+	}
+
+	expected := len(ts.GetSums())
+
+	ts.(BuilderContext).Remove("")
+	ts.(BuilderContext).Remove("Anything")
+
+	if len(ts.GetSums()) != expected {
+		t.Fatalf("Expected %v sums, go %v.", expected, ts.GetSums())
+	}
+}
+
+// Remove a tarsum (in the BuilderContext)
+func TestTarSumRemove(t *testing.T) {
+	filename := "testdata/46af0962ab5afeb5ce6740d4d91652e69206fc991fd5328c1a94d364ad00e457/layer.tar"
+	reader, err := os.Open(filename)
+	if err != nil {
+		t.Fatal(err)
+	}
+	ts, err := NewTarSum(reader, false, Version0)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Read and discard bytes so that it populates sums
+	_, err = io.Copy(ioutil.Discard, ts)
+	if err != nil {
+		t.Errorf("failed to read from %s: %s", filename, err)
+	}
+
+	expected := len(ts.GetSums()) - 1
+
+	ts.(BuilderContext).Remove("etc/sudoers")
```

Comment:
Caching
On Jun 1, 2015 18:28, "Stephen Day" <notifications@github.com> wrote:

> In pkg/tarsum/builder_context_test.go
> <https://github.com/docker/docker/pull/13630#discussion_r31477412>:
>
> > +		t.Fatal(err)
> > +	}
> > +	ts, err := NewTarSum(reader, false, Version0)
> > +	if err != nil {
> > +		t.Fatal(err)
> > +	}
> > +
> > +	// Read and discard bytes so that it populates sums
> > +	_, err = io.Copy(ioutil.Discard, ts)
> > +	if err != nil {
> > +		t.Errorf("failed to read from %s: %s", filename, err)
> > +	}
> > +
> > +	expected := len(ts.GetSums()) - 1
> > +
> > +	ts.(BuilderContext).Remove("etc/sudoers")
>
> Probably, this isn't related to this PR, but why on earth is the builder
> context integrated with the tarsum package?
>
> @vbatts <https://github.com/vbatts>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/docker/pull/13630/files#r31477412>.
>